src/app/header/header-profile/header-profile.component.ts

```diff
@@ -30,6 +46,73 @@ export class HeaderProfileComponent implements OnInit, OnDestroy {
     });
   }
 
+  private subscribeToUser(): void {
+    this.authService.user.pipe(
+      takeUntil(this.destroy$)
+    ).subscribe(user => {
+      this.user = user;
+      this.cartService.getCount();
+    });
+  }
+
+  private subscribeToProductButtonClick(): void {
+    this.authService.onProductButtonClick.pipe(
+      takeUntil(this.destroy$)
+    ).subscribe((ifToggle) => {
+        if (ifToggle) {
+          this.toggleModal();
+        }
+    });
+  }
+
+  private initForm(): void {
+    this.form = new FormGroup({
+      name: new FormControl(''),
+      email: new FormControl('', [Validators.required, Validators.email]),
+    });
+  }
+
+  public toggleUserWindow(): void {
+    this.showUserWindow = !this.showUserWindow;
+  }
+
+  public toggleModal(): void {
+    this.isActive = !this.isActive;
+    this.showUserWindow = false;
+  }
+
+  public auth(): void {
+    if (this.form.valid) {
+      const user = {
+        id: new Date().getTime().toString(),
+        ...this.form.value
+      };
+      if (this.ifRegister) {
+        this.authService.register(user).then();
+      } else {
+        this.authService.auth(this.form.value).then();
+      }
+    }
+  }
+
+  public checkAuth(): void {
+    if (!JSON.parse(localStorage.getItem('freshnesecomUser'))) {
+      this.toggleModal();
+    }
+  }
+
+  public changeTab(): void {
+    this.ifRegister = !this.ifRegister;
+  }
+
+  public logOut(): void {
+    localStorage.removeItem('freshnesecomUser');
```

Comment:
In my opinion, it is should be in service

Reply:
fixed